.github/workflows/python-app.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10.9"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest poetry
+        poetry install
+    #- name: Lint with flake8
+    #  run: |
+    #    # stop the build if there are Python syntax errors or undefined names
+    #    flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```

Comment:
Uncomment this flake8 test. When I run this test locally it passes.

Reply:
Just the one on line 34. The one on 36 does not yet pass.

Reply:
if I try to run line 34 locally I get `TypeError: missing_whitespace() takes 1 positional argument but 2 were given`

`(search) erin@Erinarxiv:~/arxiv-search$ flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/erin/.pyenv/versions/3.10.9/lib/python3.10/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/checker.py", line 82, in _mp_run
    ).run_checks()
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/checker.py", line 526, in run_checks
    self.process_tokens()
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/checker.py", line 511, in process_tokens
    self.handle_newline(token_type)
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/checker.py", line 541, in handle_newline
    self.run_logical_checks()
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/checker.py", line 448, in run_logical_checks
    for offset, text in results:
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/plugins/pycodestyle.py", line 75, in pycodestyle_logical
    yield from _missing_whitespace(logical_line, tokens)
TypeError: missing_whitespace() takes 1 positional argument but 2 were given
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/erin/.pyenv/versions/search/bin/flake8", line 8, in <module>
    sys.exit(main())
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/main/cli.py", line 23, in main
    app.run(argv)
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/main/application.py", line 198, in run
    self._run(argv)
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/main/application.py", line 187, in _run
    self.run_checks()
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/main/application.py", line 103, in run_checks
    self.file_checker_manager.run()
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/checker.py", line 235, in run
    self.run_parallel()
  File "/home/erin/.pyenv/versions/3.10.9/envs/search/lib/python3.10/site-packages/flake8/checker.py", line 204, in run_parallel
    self.results = list(pool.imap_unordered(_mp_run, self.filenames))
  File "/home/erin/.pyenv/versions/3.10.9/lib/python3.10/multiprocessing/pool.py", line 873, in next
    raise value
TypeError: missing_whitespace() takes 1 positional argument but 2 were given`

Reply:
Okay, leave that along for now then.